SimPPS/PPSPixelDigiProducer/plugins/PPSPixelDigiAnalyzer.cc

```diff
@@ -0,0 +1,309 @@
+#ifndef SimPPS_RPIXDigiAnalyzer_h
+#define SimPPS_RPIXDigiAnalyzer_h
+
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+
+#include <FWCore/Framework/interface/one/EDAnalyzer.h>
+#include <DataFormats/CTPPSDetId/interface/CTPPSPixelDetId.h>
+#include "SimDataFormats/TrackingHit/interface/PSimHitContainer.h"
+#include "SimDataFormats/TrackingHit/interface/PSimHit.h"
+#include "DataFormats/CTPPSDigi/interface/CTPPSPixelDigi.h"
+#include "DataFormats/CTPPSDigi/interface/CTPPSPixelDigiCollection.h"
+#include "Geometry/VeryForwardGeometry/interface/CTPPSPixelSimTopology.h"
+
+#include "SimDataFormats/CrossingFrame/interface/CrossingFrame.h"
+#include "SimDataFormats/CrossingFrame/interface/MixCollection.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include <FWCore/Framework/interface/Event.h>
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DataFormats/GeometryVector/interface/LocalPoint.h"
+
+#include <iostream>
+#include <string>
+
+#include "TFile.h"
+#include "TH2D.h"
+
+#define SELECTED_PIXEL_ROW 89
+#define SELECTED_PIXEL_COLUMN 23
+#define SELECTED_UNITID 2014314496
+#define TG184 0.332655724
+
+#define USE_MIDDLE_OF_PIXEL_2
+#define CENTERX 1.05
+#define CENTERY -8.475
+
+using namespace edm;
+using namespace std;
+
+class TFile;
+class PSimHit;
+
+namespace edm {
+  class ParameterSet;
+  class Event;
+  class EventSetup;
+}  // namespace edm
+
+class PPSPixelDigiAnalyzer : public edm::one::EDAnalyzer<edm::one::SharedResources> {
+public:
+  explicit PPSPixelDigiAnalyzer(const edm::ParameterSet &pset);
+  ~PPSPixelDigiAnalyzer() override;
+  void endJob() override;
+  void beginJob() override;
+  void analyze(const edm::Event &event, const edm::EventSetup &eventSetup) override;
+
+private:
+  TH2D *hAllHits;
+  TH2D *hOneHitperEvent;
+  TH2D *hOneHitperEvent2;
+  TH2D *hOneHitperEventCenter;
+  TH2D *hOneHitperEvent2Center;
+  TFile *file;
+  std::string label_;
+
+  int verbosity_;
+  edm::EDGetTokenT<edm::PSimHitContainer> psim_token;
+  edm::EDGetTokenT<edm::DetSetVector<CTPPSPixelDigi>> pixel_token;
+
+  CTPPSPixelSimTopology theRPixDetTopology_;
+  unsigned int found_corresponding_digi_count_;
+  unsigned int cumulative_cluster_size_[3];
+};
+
+PPSPixelDigiAnalyzer::PPSPixelDigiAnalyzer(const ParameterSet &pset) : theRPixDetTopology_() {
+  label_ = pset.getUntrackedParameter<string>("label");
+  verbosity_ = pset.getParameter<int>("Verbosity");
+#ifdef USE_MIDDLE_OF_PIXEL
+  file = new TFile("CTPPSPixelDigiPlots2.root", "RECREATE");
```

Comment:
@mundim why not using the https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideTFileService provided by the framework?

Reply:
@fabferro 
The same as above for SimPPS/PPSPixelDigiProducer/interface/RPixSignalPoint.h, which mean, adding this to the list of change for next releases, after the merge. We need urgently the PPS full simulation integrated.